### PR TITLE
Add `get_config_file`, etc. to provide paths without directory creation

### PR DESCRIPTION
**Pull request by philip-peterson**
This PR creates some variants on the `place_*_file` functions, which should behave the same except that they will have no side effects (directory creation).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -854,6 +884,8 @@ fn test_prefix() {
             ("HOME", format!("{}/test_files/user", cwd)),
             ("XDG_CACHE_HOME", format!("{}/test_files/user/cache", cwd)),
         ])).unwrap();
+    assert_eq!(xd.get_cache_file("cache.db"),
+        PathBuf::from(&format!("{}/test_files/user/cache/myapp/cache.db", cwd)));
```
Comment (whitequark) [inline src/lib.rs:888]: Can you add tests for the other functions too?

Comment (philip-peterson) [reply src/lib.rs:912]: Done!

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,9 @@ rust:
   - stable
 env:
   - secure: "ZmdqljmHk9oOyCJWLKWJ4ayl/7rExqss+1ZIw26zeKontCdWQ9K5NvGmbk69mD9BL07aX1O1jIaSiSrqIqV6f8/BsguADvdBUyqfdLAXQy28we+17xTphmS6vh3uT1ayqGBBhB3Qy/6JnEBp5vFdcMbwSXq02+/MwCp/wbozk4junnJ0Gg0BOK/6FUQBBnlflg1Tm5KS3pZw5H7/7GtNRW4K9ApM6EBHqBhEc8GGemCoh6q2WUabSnaxev1Bql5ZogN7wDrf5Ohpvzqv5wU25aWwRc3eJngGRelHmlGtCS1VpWSZHE4nahWj6nLIsW6Kr620Dy2ZPvNuQ/+OCC/eKNPifRlgMN2Ff1eRz9irQ6sc+KCyDmaMpAA92f1IF7y9AolGKjiTWYxKMT+0g7lbPV+WoE1+pfbsZcyOGGMLxzxxlYxSukgXN55Eqs8ag9jakjP0GtCVU3tL4nO6zyq7+UhTleWPXuTKwG64C7q3UZZUgpayzzKcJdcCdozEwuokF/kI1+o4AiF65uFw+SrQ07okJnUm86p3QhRtNSdTY6hbbLGuELQ5DbikkKoavV8WdjjqViy4tU2WafdJvA67sZvEP4h6eVvAWz+/3uZoAJyXFbfulq40XB+Fzxl/5ionnM9md86UDsX9W64dSVl+2vovj+5Jejm3wQLsuJheVG4="
+before_script:
+ - |
+   chmod 0700 test_files/user/runtime
```
Comment (whitequark) [inline .travis.yml:8]: Does this mean if I run `cargo test` it will fail? Can this file be created in Rust code?

Comment (philip-peterson) [reply .travis.yml]: Fair enough, yes, it's fixed now.

**Pull request merged**